### PR TITLE
LibJS: Update expressions and scoping improvements

### PR DESCRIPTION
**Pull request by 0xtechnobabble**
This patch introduces:

- `Let` variable parsing
- `Const` variables
- Update expressions (i.e increment and decrement)

### CODE REVIEW

**Comment by awesomekling** [comment]
Please rebase this so it has only your commits and no merge commits.

**Comment by 0xtechnobabble** [comment]
Done! Hopefully everything works, git was giving me a hard time :p

File: Libraries/LibJS/AST.cpp [hunk 1]
```diff
@@ -367,6 +367,25 @@ Value AssignmentExpression::execute(Interpreter& interpreter) const
...
+    auto previous_value = interpreter.get_variable(name);
+    ASSERT(previous_value.is_number());
+
+    switch (m_op) {
+    case UpdateOp::Increment:
+        interpreter.set_variable(name, Value(previous_value.as_double() + 1));
+        break;
+    case UpdateOp::Decrement:
+        interpreter.set_variable(name, Value(previous_value.as_double() - 1));
+    }
```
Comment (awesomekling) [inline Libraries/LibJS/AST.cpp]: Missing `break;` here.

**Comment by awesomekling** [comment]
:+1:

**Pull request merged**